ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/participantProfileServiceAgent.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Inject, Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { ConfigurationService, LoggingService, SessionMementoService, UserProfile } from 'ddp-sdk';
+import { SessionServiceAgent } from './sessionServiceAgent.service';
+
+@Injectable()
+export class ParticipantProfileServiceAgent extends SessionServiceAgent<any> {
+    constructor(
+        session: SessionMementoService,
+        @Inject('ddp.config') configuration: ConfigurationService,
+        http: HttpClient,
+        logger: LoggingService) {
+        super(session, configuration, http, logger, null);
+    }
+
+    public updateParticipantProfiles(participants: Array<{ guid: string; profile: any}>): Observable<any>[] {
+        return participants.map(participant => {
+            const {guid, profile} = participant;
+            return this.updateParticipantProfile(guid, profile);
+        });
+    }
+
+    public updateParticipantProfile(guid, profile: UserProfile): Observable<any> {
+        const profileChanges: object = {};
+        for (const key of Object.keys(profile)) {
```

Comment:
**# Optional:**
This "kind of" imperative approach looks fine in terms of space and time complexity.
I was just trying to find other workarounds, which still worth taking into consideration: 

**The long forEach() approach:**
```
 Object
      .keys(profile)
      .filter(key => profile[key])
      .forEach((key: string) => profileChanges[key] = profile[key]);
```

**The short forEach() approach (thanks to operator precedence):**
```
        Object
            .keys(profile)
            .forEach((key: string) => profile[key] && (profileChanges[key] = profile[key]));
```

**The long reduce() approach:**
```
const profileChanges: object = Object
      .keys(profile)
      .filter(key => profile[key])
      .reduce((accum: object, key: string) => ({...accum, [key]: profile[key]}), {});
```

**The short reduce() approach (thanks to operator precedence):**
```
const profileChanges: object = Object
      .keys(profile)
      .reduce((accum: object, key: string) => profile[key] && ({...accum, [key]: profile[key]}), {});
```

**As far as I tested, these approaches worked just fine. If you think there might be a scenario, in which any of these fails, please let me know.**

Reply:
Agree, it it possible. 
However, I can't see any benefit of it for that particular case.
So, let's leave it as is, shall we?

Reply:
Oki, NP